tests/causal/cate_learning/test_meta_learners.py

```diff
@@ -0,0 +1,271 @@
+
+
+def test__predict_by_treatment_flag_positive():
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1],
+            "x2": [10, 4, 15, 6],
+            TREATMENT_FEATURE: [1.0, 1.0, 0.0, 0.0],
+            "target": [1, 1, 1, 0],
+        }
+    )
+
+    learner_binary = logistic_classification_learner(
+        features=["x1", "x2", TREATMENT_FEATURE],
+        target="target",
+        params={"max_iter": 10},
+    )
+
+    predict_fn, pred_df, log = learner_binary(df)
+
+    prediction_array = _predict_by_treatment_flag(
+        df, learner_fcn=predict_fn, is_treatment=True, prediction_column="prediction"
+    )
+
+    expected_array = np.array([0.79878432, 0.65191703, 0.88361953, 0.68358276])
+
+    assert TREATMENT_FEATURE in df.columns
+    assert np.allclose(prediction_array, expected_array, atol=1e-9)
+
+
+def test__predict_by_treatment_flag_negative():
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1],
+            "x2": [10, 4, 15, 6],
+            TREATMENT_FEATURE: [1.0, 1.0, 0.0, 0.0],
+            "target": [1, 1, 1, 0],
+        }
+    )
+
+    learner_binary = logistic_classification_learner(
+        features=["x1", "x2", TREATMENT_FEATURE],
+        target="target",
+        params={"max_iter": 10},
+    )
+
+    predict_fn, pred_df, log = learner_binary(df)
+
+    prediction_array = _predict_by_treatment_flag(
+        df, learner_fcn=predict_fn, is_treatment=False, prediction_column="prediction"
+    )
+
+    expected_array = np.array([0.78981053, 0.63935056, 0.87785064, 0.67158357])
+
+    assert TREATMENT_FEATURE in df.columns
+    assert np.allclose(prediction_array, expected_array, atol=1e-9)
+
+
+def test__simulate_treatment_effect():
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1],
+            "x2": [10, 4, 15, 6],
+            "treatment": ["A", "B", "A", "control"],
+            "target": [0, 0, 0, 1],
+        }
+    )
+
+    expected = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1],
+            "x2": [10, 4, 15, 6],
+            "treatment": ["A", "B", "A", "control"],
+            "target": [0, 0, 0, 1],
+            "treatment_A__prediction_on_treatment": [0.3, 0.3, 0.0, 1.0],
+            "treatment_A__prediction_on_control": [0.2, 0.5, 0.3, 0.0],
+            "treatment_A__uplift": [0.1, -0.2, -0.3, 1.0],
+            "treatment_B__prediction_on_treatment": [0.6, 0.7, 0.0, 1.0],
+            "treatment_B__prediction_on_control": [1.0, 0.5, 1.0, 1.0],
+            "treatment_B__uplift": [-0.4, 0.2, -1.0, 0.0],
+            "uplift": [0.1, 0.2, -0.3, 1.0],
+            "suggested_treatment": [
+                "treatment_A_",
+                "treatment_B_",
+                "treatment_A_",
+                "treatment_A_",
+            ],
+        }
+    )
+
+    treatments = ["A", "B"]
+    control_name = "control"
+
+    mock_learner = create_autospec(logistic_classification_learner)
```

Comment:
I'm not sure, but I think is better to mock the entire _predict_by_treatment_flag function, which has its own unit test, and make a side effect for its output that contemplates the two necessary cases: flag = True and flag = False. We should not be restricted to the particular logistic_classification_learner object...

Reply:
you can just make a @patch for it...

Reply:
Thanks!
Just simplified the test by creating a dummy model that returns the treatment column, which will always be 1 or 0. so I could test the local variable in the function.

Reply:
I agree, and the change should be minimal.

Reply:
patching _predict_by_treatment_flag is possible but we need to patch other functions too
keeping a mock learner looks more simple, wdyt?